src/app/pages/dashboard/types/widget-component.interface.ts

```diff
@@ -27,6 +29,9 @@ type WidgetSettingsComponentType<Settings> = Settings extends SomeWidgetSettings
   ? WidgetSettingsComponent<Settings>
   : null;
 
+export type WidgetVisibilityProvider = Store;
```

Comment:
Better to specify as generics or at least unknowns. Adding visibility condition to a widget definition should not necessitate adding types here.